cmd/metrics-server/app/options/options.go

```diff
@@ -63,6 +63,23 @@ type Options struct {
 	DeprecatedCompletelyInsecureKubelet bool
 }
 
+func (o *Options) Validate() []error {
+	/*
```

Comment:
Check out this function to find some incompatible flags https://github.com/kubernetes-sigs/metrics-server/blob/master/cmd/metrics-server/app/options/options.go#L172

Flags that are incompatible will override the same field, e.g:
* `InsecureKubeletTLS`  and `InsecureKubeletTLS` as they both change `config.Client.TLSClientConfig.CAFile`
* `DeprecatedCompletelyInsecureKubelet` and all other flags as they all operate on `config.Client`

Reply:
Added two more if conditions between cert and key file and depricated-completely-insecure-kubelet